packages/cli/src/scripts/project/component/add.ts

```diff
@@ -0,0 +1,181 @@
+    })
+    .option('overwrite', {
+      requiresArg: false,
+      type: 'boolean',
+      describe: 'If true, overwrites the existing component.',
+      default: false,
+    })
+    .option('config', {
+      requiresArg: false,
+      type: 'string',
+      describe:
+        'Path to the `sitecore.cli.config` file. Supports both JavaScript (`.js`) and TypeScript (`.ts`) formats',
+    });
+}
+
+/**
+ * @param {Argv} yargs
+ */
+export function builder(yargs: Argv<AddArgs>) {
+  /* istanbul ignore next */
+  return yargs.command<AddArgs>('add <variant-id>', 'Adds a component variant', args, handler);
+}
+
+/**
+ * Handler for the add command.
+ * @param {AddArgs} argv - The arguments passed to the command.
+ */
+export async function handler(argv: AddArgs) {
+  const { variantId, targetPath: targetPathArg, skipComponentMap, config, overwrite } = argv;
+
+  console.log(chalk.green('Adding component variant'));
+
+  let targetPath = targetPathArg;
+
+  const cliConfig = loadCliConfig(config);
+
+  if (!cliConfig.config) {
+    console.error(
+      'The `sitecore.cli.config` file is missing a `config`. Please add it to use this command.'
+    );
+    return;
+  }
+
+  const { contextId, edgeUrl } = cliConfig.config.api.edge;
+
+  try {
+    const spec = await getComponentVariantSpec({
+      edgeUrl,
+      variantId,
+      targetPath,
+    });
+
+    const componentType = spec.meta['contentsdk-component-type'];
+    const componentName = spec.meta['contentsdk-component-name'];
+    const variantName = spec.meta['contentsdk-component-variant-name'];
+    const title = spec.title;
+
+    if (componentType !== 'variant') {
+      console.error(
+        chalk.red(
+          `The component "${title}" is not a content-sdk variant. Please, select a content-sdk variant to use this command.`
+        )
+      );
+
+      return;
+    }
+
+    if (!targetPath) {
+      const { paths, exclude } = cliConfig.componentMap;
+
+      const components = getComponentList(paths, exclude, true);
+
+      const registeredComponent = components.find((c) => c.componentName === componentName);
+
+      if (registeredComponent) {
+        targetPath = registeredComponent.filePath
+          .replace(/\\/g, '/')
+          .replace(/([^/]+)\.([^.]+)$/, `$1.${variantName}.$2`);
+      } else {
+        targetPath = await inquirer
+          .prompt({
+            type: 'input',
+            name: 'targetPath',
+            required: true,
+            message: `Enter the target path for the component variant.\nThe filename must follow the format: {componentName}.{variantName}.{extension}\n(example: src/components/MyComponent/MyComponent.variantA.ts):`,
+          })
+          .then((answer) => answer.targetPath);
+      }
+    }
+
+    const registryUrl = getComponentRegistryUrl({
+      variantId,
+      contextId,
+      targetPath: targetPath as string,
+    });
+
+    execSync(`npx shadcn@latest add ${registryUrl}${overwrite ? ' --overwrite' : ''}`, {
```

Comment:
I think we should pin certain version of shadcn, as latest could take a new major version our call is not compatible with